### PR TITLE
[OLEAUT32] Implement call_method for x64 MSVC.

### PR DESCRIPTION
**Pull request by ThFabba**
Code should be identical to https://github.com/reactos/reactos/blob/master/dll/win32/oleaut32/typelib.c#L6450-L6488

### CODE REVIEW

File: dll/win32/oleaut32/msvc.S [hunk 1]
```diff
@@ -13,9 +13,42 @@
...
+    push rdi
+    .PUSHREG rdi
+    .ENDPROLOG
+
+    mov rax, rcx
+    mov rcx, 4
+    cmp rdx, rcx
+    cmovg rcx, rdx
+    lea rdx, [rcx * 8]
+    sub rsp, rdx
```
Comment (tkreuzer) [inline dll/win32/oleaut32/msvc.S]: Usually this requires a call to __chkstk. I guess we can live without, if we assume that this function can never be called with any parameter count that makes the total stack allocation exceed 4k.

File: dll/win32/oleaut32/msvc.S [hunk 2]
```diff
@@ -13,9 +13,42 @@
 .code64
 
 PUBLIC call_method
-call_method:
-    int 2ch
+FUNC call_method
+    push rbp
+    .PUSHREG rbp
+    mov rbp, rsp
```
[the remark this answers is not in the document]
Comment (tkreuzer) [reply dll/win32/oleaut32/msvc.S]:
```suggestion
```

File: dll/win32/oleaut32/msvc.S [hunk 3]
```diff
@@ -13,9 +13,42 @@
...
+    push rsi
+    .PUSHREG rsi
+    push rdi
+    .PUSHREG rdi
+    .ENDPROLOG
```
Comment (tkreuzer) [inline dll/win32/oleaut32/msvc.S]:
```suggestion
    mov rbp, rsp
    .SETFRAME rbp
    .ENDPROLOG
```

File: dll/win32/oleaut32/msvc.S [hunk 4]
```diff
@@ -13,9 +13,42 @@
...
+    mov rdx, qword ptr [rsp + 8]
+    mov r8, qword ptr [rsp + 16]
+    mov r9, qword ptr [rsp + 24]
+    movq xmm0, qword ptr [rsp + 0]
+    movq xmm1, qword ptr [rsp + 8]
+    movq xmm2, qword ptr [rsp + 16]
+    movq xmm3, qword ptr [rsp + 24]
+    call qword ptr [rax]
+
+    lea rsp, [rbp - 16]
```
Comment (tkreuzer) [inline dll/win32/oleaut32/msvc.S]:
```suggestion
    lea rsp, [rbp]
```

**Comment by ThFabba** [comment]
I'm happy to make changes to the directives, but the instructions aren't mine -- they're just translated from Wine's GCC version: https://github.com/reactos/reactos/blob/master/dll/win32/oleaut32/typelib.c#L6448-L6488
So unless they're properly broken, I'd prefer to keep them in sync rather than rolling our own.

**Comment by Mouvedia** [comment]
Should this issue be added to the x64 bringup milestone?

**Comment by tkreuzer** [comment]
> 
> So unless they're properly broken, I'd prefer to keep them in sync rather than rolling our own.

It is broken. I'd prefer to have it fixed for GCC as well, or generally use this file instead of the inline asm.

**Comment by ThFabba** [comment]
Superseded by 38fc6e1274c979de54089ce825a1f2d0c3b50cfe

**Pull request closed without merging**